### ISSUE TITLE
[Feature] Output tokenized example for debugging during training

### ISSUE THREAD

**Issue by jgreer013** [oumi-ai/oumi#1369]
### Feature request

When running training, it'd be useful to log a single example from the dataset that has been tokenized for debugging.

For example:
1. Take the first example or batch in the training set (peek the top of the dataloader?)
2. Print all the following:
a. Raw Example
b. Formatted Example
c. Tokenized Example (list of tuples of <token_id, decoded_token>)
d. Model Input (tokens with attention mask and labels)

Ideally this should either be a debug log or its output explicitly controlled by a flag.

Likely, all parts except the model input can be done when loading the dataset. The model input will need to be done inside the collator(s).

### Motivation / references

Concrete example using `HuggingFaceTB/SmolLM2-1.7B-Instruct` tokenizer, with model setup to be trained on **completions only**:

Raw Example
```
[
  {'role': 'system', 'content': 'This is a system instruction.'},
  {'role': 'user', 'content': 'This is a user instruction'},
  {'role': 'assistant', 'content': 'This is an assistant response.'},
]
```

Formatted Example
```
<|im_start|>system
This is a system instruction.<|im_end|>
<|im_start|>user
This is a user instruction<|im_end|>
<|im_start|>assistant
This is an assistant response.<|im_end|>
```

Tokenized Example
```
[(1, '<|im_start|>'), (9690, 'system'), (198, '\n'), (1348, 'This'), (314, ' is'), (253, ' a'), (817, ' system'), (5785, ' instruction'), (30, '.'), (2, '<|im_end|>'), (198, '\n'), (1, '<|im_start|>'), (4093, 'user'), (198, '\n'), (1348, 'This'), (314, ' is'), (253, ' a'), (2914, ' user'), (5785, ' instruction'), (2, '<|im_end|>'), (198, '\n'), (1, '<|im_start|>'), (520, 'ass'), (9531, 'istant'), (198, '\n'), (1348, 'This'), (314, ' is'), (354, ' an'), (11173, ' assistant'), (2426, ' response'), (30, '.'), (2, '<|im_end|>'), (198, '\n')]
```

Model Input Example
```
{
  'input': [1, 9690, 198, 1348, 314, 253, 817, 5785, 30, 2, 198, 1, 4093, 198, 1348, 314, 253, 2914, 5785, 2, 198, 1, 520, 9531, 198, 1348, 314, 354, 11173, 2426, 30, 2, 198],
  'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
  'labels': [-100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, -100, 1348, 314, 354, 11173, 2426, 30, 2, 198],
}
```

### Your contribution

Example code:
```
import transformers
tokenizer = transformers.AutoTokenizer.from_pretrained('HuggingFaceTB/SmolLM2-1.7B-Instruct')

chat = [
  {'role': 'system', 'content': 'This is a system instruction.'},
  {'role': 'user', 'content': 'This is a user instruction'},
  {'role': 'assistant', 'content': 'This is an assistant response.'},
]

formatted = tokenizer.apply_chat_template(chat, tokenize=False)
print(formatted)

tokens = tokenizer.apply_chat_template(chat)
decoded = [tokenizer.decode(t) for t in tokens]
print(list(zip(tokens, decoded)))
```

**Comment by aniruddh-alt** [comment]
Hi @nikg4, I would like to work on this issue if it is still active!

**Comment by nikg4** [comment]
@aniruddh-alt  of course!
